vignettes/markdown.Rmd

```diff
@@ -1,17 +1,188 @@
-<!--
-%\VignetteEngine{knitr::knitr}
-%\VignetteIndexEntry{Write R documentation in Markdown}
--->
+---
+title: "Write R Documentation in Markdown"
+author: "Gábor Csárdi"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Write R Documentation in Markdown}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
 
-```{r, echo = FALSE, message = FALSE}
-knitr::opts_chunk$set(
-  comment = "#>",
-  error = FALSE,
-  tidy = FALSE
-)
+```{r, include = FALSE}
+knitr::opts_chunk$set(comment = "#>", collapse = TRUE)
 ```
 
-# Write R documentation in Markdown
+# Introduction
 
-This is a vignette to be written once we know what exactly the Markdown
-support will cover.
+Starting from version 6.0.0, roxygen supports markdown markup within most roxygen tags. Roxygen uses the `commonmark` package, which is based on the CommonMark Reference Implementation to parse these tags. See http://commonmark.org/help/ for more about the parser and the markdown language it supports.
+
+# Turning on markdown support
+
+Currently you have to turn on markdown formatting manually for each roxygen chunk. You can do this by using the `@md` tag anywhere in the chunk. In the future roxygen will provide a global switch that turns on markdown formatting for the whole package. (And also a `@noMd` to turn it off, optionally.)
+
+Here is an example roxygen chunk that uses markdown.
+
+```r
+#' Use roxygen to document a package.
+#'
+#' This function is a wrapper for the [](roxygen2::roxygenize) function from 
```

Comment:
Maybe the best alternative is the one that @hadley suggested in that thread: use bracket links, and provide the references to commonmark. E.g. this works:

```
This will be a link here: [roxygen2::roxygenize].

[roxygen2::roxygenize]: blah
```

and we would provide the reference definitions at the end, the user would not have to type them. Not sure how, but we could pre-parse the text, and stick there everything we could find?